### PR TITLE
Add kubectl-config-cleanup plugin manifest

### PR DESCRIPTION
**Pull request by dbkegley**
-----

**Checklist for plugin developers:**

- [x] Read the [Plugin Naming Guide](https://github.com/GoogleContainerTools/krew/tree/master/docs/NAMING_GUIDE.md) (for new plugins)
- [x] Verify the installation from URL or a local archive works (`kubectl krew install --manifest=[...] --archive=[...]`)

### CODE REVIEW

File: plugins/config-cleanup.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: config-cleanup
+spec:
+  version: "v0.2.0"
+  shortDescription: Automatically clean up your kubeconfig
```
Comment (ahmetb) [inline plugins/config-cleanup.yaml]: Can you add a long description field?

Why is it automatic? How does it clean up? What parts does it cleanup?

Ideally these should be answered without installing the plugin.

Comment (dbkegley) [reply plugins/config-cleanup.yaml]: I've updated the manifest with a longDescription

File: plugins/config-cleanup.yaml [hunk 2]
```diff
@@ -6,6 +6,7 @@ metadata:
 spec:
   version: "v0.2.0"
   shortDescription: Automatically clean up your kubeconfig
+  longDescription: Every cloud provider has their own utilities for adding kubernetes cluster credentials to your kubeconfig but most don't offer the ability to clean it up once the cluster is deleted. For those of us who launch and delete multiple clusters per day, it is useful to have an automated way to clean up old kubeconfig entries. This plugin will attempt to connect to each cluster defined in a context, if the connection succeeds then the user, cluster, and context entries are maintained in the result. Otherwise, the entries are removed.
```
Comment (ahmetb) [inline plugins/config-cleanup.yaml]: This should be “description”

Also please do line wrapping.

Comment (dbkegley) [reply plugins/config-cleanup.yaml]: Updated field name, fixed line wrapping, and shortened the description

File: plugins/config-cleanup.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (ahmetb) [inline plugins/config-cleanup.yaml]: no need for this

File: plugins/config-cleanup.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: config-cleanup
+spec:
+  version: "v0.2.0"
+  shortDescription: Automatically clean up your kubeconfig
+  description: This plugin will attempt to connect to each apiserver specified by a context entry in your kubeconfig.
```
[the remark this answers is not in the document]
Comment (dbkegley) [reply plugins/config-cleanup.yaml]: Fixed multi-line str and line length. Anything else I should add/remove in the desc?

**Comment by ahmetb** [comment]
I think you also need to sign the Linux Foundation CLA as the bot has instructed above. Let me know if you already did.

**Comment by ahmetb** [comment]
I just ran this plugin and it cleared up all my contexts:

```
kubectl config-cleanup
apiVersion: v1
clusters: []
contexts: []
current-context: ""
kind: Config
preferences: {}
users: []
```

The whole cmd took 0.1 secs. I don't think it's working correctly. I have 4 GKE clusters in my context. All of them are actually working fine.

**Comment by dbkegley** [comment]
> I think you also need to sign the Linux Foundation CLA as the bot has instructed above. Let me know if you already did.

Still waiting on approval for the CLA, I'll update once it's approved.

> I don't think it's working correctly. I have 4 GKE clusters in my context. All of them are actually working fine.

I just tested on GKE and am seeing the same behavior, we can close this PR for now. I'll have to make some additional changes before this plugin is ready for general use. Thank you for the initial review. I'll reopen once the CLA is approved and I've tested the plugin more thoroughly with each of the auth-provider plugins.

**Comment by ahmetb** [comment]
It would be good to keep it open so that we don't lose context on the review history.

**Comment by ahmetb** [comment]
/hold
until CLA is signed, and until it works with auth providers like GKE.

**Comment by dbkegley** [comment]
The newest version of the plugin should work with all auth provider plugins, please let me know if you run into any issues

**Comment by ahmetb** [comment]
@dbkegley just tried, it has removed unreachable context, but I still see a bunch of dangling (unused) `users` and `clusters` entries, is this expected/by design?

**Comment by ahmetb** [comment]
My bad, just saw `--users` and `--clusters`. But when I used it, I got error:

```
$ kubectl config-cleanup --users --clusters > kubeconfig

E0411 16:31:00.594794   59322 round_trippers.go:174] CancelRequest not implemented by *gcp.conditionalTransport
```

but `kubeconfig` was written to the file successfully.

**Comment by dbkegley** [comment]
I've seen that one before as well, I believe it's related to this issue https://github.com/kubernetes/kubernetes/issues/73791

I've seen it from both the exec plugin and the gcp auth plugins

**Comment by ahmetb** [comment]
Got it. The printed error is not a blocker. Next, please look at signing the CLA. 
/hold cancel

**Comment by dbkegley** [comment]
I signed it

**Pull request merged**